mindsdb/api/http/namespaces/knowledge_bases.py

```diff
@@ -329,86 +411,45 @@ def post(self, project_name, knowledge_base_name):
             )
 
-            logger.error('Missing parameter "query" in POST body')
-
-        llm_model = request.json.get('llm_model')
-        if llm_model is None:
-            logger.warn(f'Missing parameter "llm_model" in POST body, using default llm_model {DEFAULT_LLM_MODEL}')
-
-        prompt_template = request.json.get('prompt_template')
-        if prompt_template is None:
-            logger.warn(f'Missing parameter "prompt_template" in POST body, using default prompt template {DEFAULT_RAG_PROMPT_TEMPLATE}')
-
-        session = SessionController()
         project_controller = ProjectController()
         try:
             project = project_controller.get(name=project_name)
         except EntityNotExistsError:
             # Project must exist.
+            logger.error("Project not found, please check the project name exists")
             return http_error(
                 HTTPStatus.NOT_FOUND,
                 'Project not found',
                 f'Project with name {project_name} does not exist'
             )
-            logger.error("Project not found, please check the project name exists")
 
+        session = SessionController()
         # Check if knowledge base exists
         table = session.kb_controller.get_table(knowledge_base_name, project.id)
         if table is None:
+            logger.error("Knowledge Base not found, please check the knowledge base name exists")
             return http_error(
                 HTTPStatus.NOT_FOUND,
                 'Knowledge Base not found',
                 f'Knowledge Base with name {knowledge_base_name} does not exist'
             )
 
-            logger.error("Knowledge Base not found, please check the knowledge base name exists")
-
-        # Get retrieval config, if set
-        retrieval_config = request.json.get('retrieval_config', {})
-        if not retrieval_config:
-            logger.warn('No retrieval config provided, using default retrieval config')
-
-        # add llm model to retrieval config
-        if llm_model is not None:
-            retrieval_config['llm_model_name'] = llm_model
-
-        # add prompt template to retrieval config
-        if prompt_template is not None:
-            retrieval_config['rag_prompt_template'] = prompt_template
-
-        # add llm provider to retrieval config if set
-        llm_provider = request.json.get('model_provider')
-        if llm_provider is not None:
-            retrieval_config['llm_provider'] = llm_provider
-
-        # build rag pipeline
-        rag_pipeline = table.build_rag_pipeline(retrieval_config)
-
-        # get response from rag pipeline
-        rag_response = rag_pipeline(query)
-        response = {
-            'message': {
-                'content': rag_response.get('answer'),
-                'context': rag_response.get('context'),
-                'role': 'assistant'
-            }
-        }
-
-        return response
+        completion_type = request.json.get('type', 'chat')
```

Comment:
could be good to add this behaviour to the doc string: "unless specified otherwise by default completion_type is chat"